app/models/contract.rb

```diff
@@ -4,7 +4,16 @@ class Contract < ApplicationRecord
   accepts_nested_attributes_for :documents
   belongs_to :created_by, class_name: "User"
 
+  belongs_to :supplier, optional: true
+  belongs_to :affiliate, optional: true
+
   validates :title, presence: true
+  validates :supplier_id, presence: true, if: :status_is_approved?
+  validates :supplier_number, presence: true, if: :status_is_approved?
+  validates :affiliate_name, presence: true, if: :status_is_approved?
```

Comment:
We shouldn't check the validation of affiliate/supplier in Contract. Move this to `models/affiliate.rb` and `models/supplier.rb` respectively.